src/cogs/snap/snap.py

```diff
@@ -0,0 +1,30 @@
+import io
+import os
+import discord
+from discord.ext import commands
+from discord import app_commands, File
+from dotenv import load_dotenv
+
+load_dotenv()
+
+
+class Snap(commands.Cog):
+
+    def __init__(self, _bot: commands.Bot):
+        self.bot: commands.Bot = _bot
+
+    @app_commands.command(name='snap', description='Snapshot a text channel')
+    @app_commands.guilds(int(os.getenv('TEST_GUILD')))
+    async def snap(self, interaction: discord.Interaction, channel: discord.TextChannel = None, length: int = 200):
+        messages = await channel.history(limit=length).flatten()
+        message_return = []
+        for message in messages:
+            message_return.append(f"{message.created_at} : {message.author.name} : {message.content}")
+
+        message_return.reverse()
```

Comment:
we could loop through this backwards instead of enumerating it twice, unless .reverse isn't o(n) somehow

Reply:
Addressed by using `reversed()` on the iterable